src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -728,68 +736,137 @@ impl<'a> ImmutableBuffer<'a> {
+                (header_length, remaining_input)
             }
             9 => {
-                let (flex_uint, remaining_input) = input_after_opcode.read_flex_uint()?;
-                let sequence = EncodedAnnotations {
-                    encoding: AnnotationsEncoding::FlexSym,
-                    header_length: u8::try_from(1 + flex_uint.size_in_bytes()).map_err(|_| {
-                        IonError::decoding_error("found a 256+ byte annotations header")
-                    })?,
-                    sequence_length: u16::try_from(flex_uint.value()).map_err(|_| {
-                        IonError::decoding_error(
-                            "the maximum supported annotations sequence length is 65KB.",
-                        )
-                    })?,
-                };
-                (
-                    sequence,
-                    remaining_input.consume(sequence.sequence_length as usize),
-                )
+                let (flex_uint, input_after_header) = input_after_opcode.read_flex_uint()?;
+                let sequence_length = flex_uint.value() as usize;
+                if input_after_header.len() < sequence_length {
+                    return IonResult::incomplete(
+                        "reading an annotations sequence",
+                        input_after_header.offset(),
+                    );
+                }
+                // Header = 0xE6, followed by FlexUInt size in bytes
+                let header_length = u8::try_from(1 + flex_uint.size_in_bytes()).map_err(|_| {
+                    IonError::decoding_error("found a 256+ byte annotations header")
+                })?;
+                let remaining = input_after_header.consume(sequence_length);
+                (header_length, remaining)
             }
             _ => unreachable!("reading flexsym annotations sequence with invalid length code"),
         };
+
+        let sequence_length =
+            u16::try_from(remaining_input.offset() - (self.offset() + header_length as usize))
+                .map_err(|_| {
+                    IonError::decoding_error(
+                        "the maximum supported annotations sequence length is 65KB",
+                    )
+                })?;
+
+        let sequence = EncodedAnnotations {
+            encoding: AnnotationsEncoding::FlexSym,
+            header_length,
+            sequence_length,
+        };
+
         Ok((sequence, remaining_input))
     }
 
+    /// Skips beyond a `FlexSym` at the head of the buffer, returning the remaining slice.
+    fn consume_flex_sym(self) -> IonResult<Self> {
+        // TODO: As an optimization, see if we can avoid actually reading the flex_sym.
+        let (flex_sym, remaining) = self.read_flex_sym()?;
+        if let FlexSymValue::Opcode(opcode) = flex_sym.value() {
+            todo!("FlexSym escapes in annotation sequences; opcode: {opcode:?}");
```

Comment:
🗺️ The skipper doesn't (yet) attempt to interpret the opcode and skip whatever it indicates.